vhost-user-backend/src/event_loop.rs

```diff
@@ -134,14 +200,56 @@ where
         }
     }
 
-    pub(crate) fn register_event(&self, fd: RawFd, ev_type: EventSet, data: u64) -> Result<()> {
-        self.epoll
-            .ctl(ControlOperation::Add, fd, EpollEvent::new(ev_type, data))
+    pub(crate) fn register_event(&self, fd: RawFd, ev_type: Interest, data: u64) -> Result<()> {
+        self.registry
+            .register(&mut SourceFd(&fd), Token(data as usize), ev_type)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
```

Comment:
From https://docs.rs/mio/latest/mio/struct.Registry.html#method.register:

> Callers must ensure that if a source being registered with a Poll instance was previously registered with that Poll instance, then a call to [deregister](https://docs.rs/mio/latest/mio/struct.Registry.html#method.deregister) has already occurred. Consecutive calls to register is unspecified behavior.

So maybe we can't do that, without calling `deregister`

Reply:
Maybe we need to call `reregister`, see https://docs.rs/mio/latest/mio/struct.Registry.html#method.reregister

> Re-registering an event source allows changing the details of the registration. Specifically, it allows updating the associated token and interests specified in previous register and reregister calls.